### PR TITLE
Storage: Optionally overwrite existing files

### PR DESCRIPTION
**Pull request by adela-almasan**
https://user-images.githubusercontent.com/88068998/178343885-580e1f83-a941-4cf3-aa0f-5714fde8f32a.mov

### CODE REVIEW

**Review by ryantxu — APPROVED** [review]
🌮

File: public/app/features/storage/UploadView.tsx [hunk 1]
```diff
@@ -84,8 +88,25 @@ export const UploadView = ({ folder, onUpload }: Props) => {
...
+          </Alert>
+        </div>
+      )}
+
       <ButtonGroup>
-        <Button className={styles.button} variant={'primary'} disabled={!file} onClick={doUpload}>
+        <Button
+          className={styles.button}
+          variant={'primary'}
+          disabled={!file || (filenameExists && !overwriteExistingFile)}
```
Comment (nmarrs) [inline public/app/features/storage/UploadView.tsx:107]: Nit but may be a bit cleaner to separate this out into a separate bool above the return, maybe something like `isUploadDisabled`

**Review by nmarrs — APPROVED** [review]
Tested locally and works great. A+ for demo picture selection :)

**Comment by ryantxu** [comment]
> A+ for demo picture selection :)
A+ adorable

**Pull request merged**